deploy_vpipe.sh

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env bash
+
+# defaults
+PREFIX=$(pwd)
+FORCE=
+BRANCH=master
+RELEASE=
+WORKDIR=
+MINIMAL=
+
+# Helper
+fail() {
+	printf '\e[31;1mArgh: %s\e[0m\n'	"$1"	1>&2
+	[[ -n "$2" ]] && echo "$2" 1>&2
+	exit 1
+}
+
+oops() {
+	printf '\e[33;1mOops:%s\e[0m\n'	"$1"	 1>&2
+}
+
+title() {
+	printf '\e[34;1m======================\n%s\n======================\e[0m\n\n'	"$1"
+}
+
+message() {
+	printf '\e[37;1m%s\t%s\e[0m\n' "$1" "$2"
+}
+
+status() {
+	printf '\e[36;1m%s\e[0m\n'	"$1"
+}
+
+check_directory() {
+	if [[ -d "$1" ]]; then
+		if (( FORCE )); then
+			rm -rf "$1"
+		else
+			fail "${2:-Directory} ${1} already exists" 'check and use -f if needed'
+		fi
+	fi
+}
+
+usage() {
+echo "usage: $0 [options]
+options:
+-f           force overwriting directories
+-p PREFIX    prefix directory under which to install V-pipe
+             [default: current directory]
+-b BRANCH    install specified branch of V-pipe${BRANCH:+
+             [default: $BRANCH]}
+-r RELEASE   install specified release package of V-pipe${RELEASE:+
+             [default: $RELEASE]}
+-w WORKDIR   create and populate working directory
+-m           only minimal working directory
+-h           print this help message and exit"
+}
+
+
+# parameters
+while getopts ":fp:b:r:w:mh" opt; do
+	case ${opt} in
+		f)
+			FORCE=1
+		;;
+		p)
+			PREFIX="${OPTARG}"
+		;;
+		b)
+			BRANCH="${OPTARG}"
+		;;
+		r)
+			RELEASE="${OPTARG}"
+		;;
+		w)
+			WORKDIR="${OPTARG}"
+		;;
+		m)
+			MINIMAL='-m'
+		;;
+		h)
+			usage
+			exit 0
+		;;
+		\?)
+			fail "Invalid option: ${OPTARG}"
+		;;
+		:)
+			fail "Invalid option: ${OPTARG} requires an argument"
+		;;
+	esac
+done
+shift $((OPTIND -1))
+
+[[ -z "${BRANCH}" && -z "${RELEASE}" ]] && fail "Please specify either a branch or a release package"
+
+
+###################
+#                 #
+#   Basic stuff   #
+#                 #
+###################
+
+DOWNLOAD=
+if [[ -x $(which wget) ]] || wget --version; then # most Linux distros (Gentoo)
+	DOWNLOAD='wget'
+elif [[ -x $(which curl) ]] || curl --version; then # a few Linux distros (CentOS, Archlinux) and Mac OS X
+	DOWNLOAD='curl -O'
+else
+	fail 'Please install either wget or curl'
+fi;
+
+# HACK we will install git from conda if not available locally
+GIT=
+if [[ -x $(which git) ]] && git --version 2>/dev/null >/dev/null; then # most computers with development tools installed, clusters, etc
+	GIT=
+else # out-of-the box Mac OS X and some Linux dockers
+	oops 'git is missing, I will download it from conda'
+	GIT=git
+fi;
+
+# BUG having environment modifier (such as conda or modules) is a *horrendously bad idea* that can cause hard to undertsand errors much later
```

Comment:
typo: modifier**s**
typo: understand
s/BUG/CHECK/

Also, I do not think it is a bad idea, although it clearly would break this script. So, perhaps the comment is a bit misleading in that regard.